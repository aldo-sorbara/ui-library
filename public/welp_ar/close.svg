<svg xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    height="24"
    width="24"
    viewBox="0 0 35 35">
  <path d="M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z"/>
</svg>
