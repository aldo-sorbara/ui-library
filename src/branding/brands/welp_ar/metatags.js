export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
