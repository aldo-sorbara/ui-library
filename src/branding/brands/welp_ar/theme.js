import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
