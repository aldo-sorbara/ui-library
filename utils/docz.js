export const styles = {
  display: 'flex',
  flex: 1,
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'space-around'
};
