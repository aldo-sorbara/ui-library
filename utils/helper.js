import { shallow, mount } from 'enzyme';

import { ThemeConsumer } from 'styled-components';

export const shallowWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return shallow(children);
};

export const mountWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return mount(children);
};
