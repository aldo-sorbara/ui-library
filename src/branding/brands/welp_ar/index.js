import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
