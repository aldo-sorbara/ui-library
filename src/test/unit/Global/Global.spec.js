import React from 'react';
import {
  Container,
  Wrapper,
  Headline,
  Description,
  Legales,
  Label,
  LabelLarge,
  SecondHeadline,
  ContainerDesktop,
  Desktop,
  Onboarding,
  Terms,
  H1,
  SectionDescription,
  UrlLink,
  Timer,
  DesktopText,
  Bind,
  Logo,
  LogoMobile,
  TextPromotion
} from '../../../components/web/global';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../../branding/brands/welp_ar';
import 'jest-styled-components';

describe('Container component', () => {
  it('should render container snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Container className='test-container' data='container'>
          This is a container
        </Container>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render absolute container', () => {
    const wrapper = mountWithTheme(
      <Container className='test-container' absolute data='container-test'>
        Container
      </Container>,
      welp_ar.theme
    );
    expect(wrapper.props().absolute).toBe(true);
    expect(wrapper).toHaveStyleRule('position', 'absolute');
  });
});

describe('Wrapper component', () => {
  it('should render Wrapper snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Wrapper className='test-wrapper' data='wrapper'>
          This is a wrapper
        </Wrapper>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Headline component', () => {
  it('should render Headline snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Headline className='test-headline' data='headline'>
          This is a headline
        </Headline>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Description component', () => {
  it('should render Description snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Description className='test-description' data='description'>
          This is a description
        </Description>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Legales component', () => {
  it('should render Legales snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Legales className='test-legales' data='legales'>
          This is a legales
        </Legales>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Label component', () => {
  it('should render Label snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Label className='test-label' data='label'>
          This is a label
        </Label>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('LabelLarge component', () => {
  it('should render LabelLarge snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <LabelLarge className='test-labelLarge' data='label-large'>
          This is a label large
        </LabelLarge>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('SecondHeadline component', () => {
  it('should render SecondHeadline snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <SecondHeadline className='test-secondHeadline' data='second-headline'>
          This is a Second Headline
        </SecondHeadline>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ContainerDesktop component', () => {
  it('should render ContainerDesktop snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <ContainerDesktop
          className='test-containerDesktop'
          data='container-desktop'
        >
          This is a ContainerDesktop component
        </ContainerDesktop>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Desktop component', () => {
  it('should render Desktop snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Desktop className='test-desktop' data='desktop' />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render Desktop with background', () => {
    const wrapper = mountWithTheme(
      <Desktop className='test-desktop' data='desktop-test' img='src/test' />,
      welp_ar.theme
    );
    expect(wrapper.props().img).toBe('src/test');
    expect(wrapper).toHaveStyleRule('background-image', 'url(src/test)');
  });
});

describe('Onboarding component', () => {
  it('should render Onboarding snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Onboarding className='test-onboarding' data='onboarding'>
          This is an Onboarding component
        </Onboarding>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render Onboarding with overflow-y: hidden', () => {
    const wrapper = mountWithTheme(
      <Onboarding className='test-onboarding' data='onboarding-test' isLead>
        This is an Onboarding component
      </Onboarding>,
      welp_ar.theme
    );
    expect(wrapper.props().isLead).toBe(true);
    expect(wrapper).toHaveStyleRule('overflow-y', 'hidden');
  });
});

describe('Terms component', () => {
  it('should render Terms snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Terms className='test-terms' data='terms'>
          This is a Terms component
        </Terms>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('H1 component', () => {
  it('should render H1 snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <H1 className='test-h1' data='h1'>
          This is an H1 component
        </H1>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('SectionDescription component', () => {
  it('should render SectionDescription snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <SectionDescription
          className='test-sectionDescription'
          data='section-description'
        >
          This is a SectionDescription component
        </SectionDescription>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render SectionDescription with bigger font', () => {
    const wrapper = mountWithTheme(
      <SectionDescription
        className='test-sectionDescription'
        data='section-description-test'
        bigger
      >
        This is a SectionDescription component
      </SectionDescription>,
      welp_ar.theme
    );
    expect(wrapper.props().bigger).toBe(true);
    expect(wrapper).toHaveStyleRule('font-size', '1.3rem');
    expect(wrapper).toHaveStyleRule('margin-bottom', '50px');
    expect(wrapper).toHaveStyleRule('margin-top', '-20px');
  });
});

describe('UrlLink component', () => {
  it('should render UrlLink snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <UrlLink className='test-urlLink' data='url-link'>
          This is an UrlLink component
        </UrlLink>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Timer component', () => {
  it('should render Timer snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Timer className='test-timer' data='timer'>
          This is a Timer component
        </Timer>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('DesktopText component', () => {
  it('should render DesktopText snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <DesktopText className='test-desktopText' data='desktop-text'>
          This is a DesktopText component
        </DesktopText>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Bind component', () => {
  it('should render Bind snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Bind className='test-bind' data='bind' />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Logo component', () => {
  it('should render Logo snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Logo className='test-logo' data='logo' />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('LogoMobile component', () => {
  it('should render LogoMobile snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <LogoMobile className='test-logo-mobile' data='logo-mobile' />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('TextPromotion component', () => {
  it('should render TextPromotion snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <TextPromotion className='test-text-promotion' data='text-promotion'>
          This is a TextPromotion component
        </TextPromotion>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
