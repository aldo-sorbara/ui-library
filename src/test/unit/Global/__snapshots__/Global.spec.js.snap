// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bind component should render Bind snapshot 1`] = `
.c0 {
  background: linear-gradient(to Left,rgba(0,0,0,0.4),rgba(0,0,0,0));
  height: 100%;
  position: absolute;
  right: 0;
  width: 3rem;
}

<div
  className="test-bind c0"
  data="bind"
/>
`;

exports[`Container component should render container snapshot 1`] = `
.c0 {
  background-color: transparent;
  margin: 0 auto;
  max-width: unset;
}

<div
  className="test-container c0"
  data="container"
>
  This is a container
</div>
`;

exports[`ContainerDesktop component should render ContainerDesktop snapshot 1`] = `
.c0 {
  background: linear-gradient( 135deg, #0082ff 0%, #3a66ab 80% );
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  overflow: hidden;
  position: absolute;
  width: 100%;
}

<div
  className="test-containerDesktop c0"
  data="container-desktop"
>
  This is a ContainerDesktop component
</div>
`;

exports[`Description component should render Description snapshot 1`] = `
.c0 {
  margin-bottom: 1rem;
  margin-top: 0;
  font-size: .875rem;
  font-weight: 400;
  line-height: 1.25;
}

<p
  className="test-description c0"
  data="description"
>
  This is a description
</p>
`;

exports[`Desktop component should render Desktop snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-size: cover;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  width: 100%;
}

<div
  className="test-desktop c0"
  data="desktop"
/>
`;

exports[`DesktopText component should render DesktopText snapshot 1`] = `
.c0 {
  text-align: center;
}

<div
  className="test-desktopText c0"
  data="desktop-text"
>
  This is a DesktopText component
</div>
`;

exports[`H1 component should render H1 snapshot 1`] = `
.c0 {
  color: #ffffff;
  font-size: 36px;
  margin-bottom: 64px;
  margin-top: 0;
  font-size: 1.625rem;
  font-weight: 700;
  line-height: 1.14;
  position: relative;
  z-index: 2;
}

<h1
  className="test-h1 c0"
  data="h1"
>
  This is an H1 component
</h1>
`;

exports[`Headline component should render Headline snapshot 1`] = `
.c0 {
  margin-bottom: .5rem;
  margin-left: 0;
  margin-top: 0;
  font-size: 1.625rem;
  font-weight: 700;
  line-height: 1.14;
}

<h1
  className="test-headline c0"
  data="headline"
>
  This is a headline
</h1>
`;

exports[`Label component should render Label snapshot 1`] = `
.c0 {
  margin-bottom: .25rem;
  margin-top: 20;
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
}

<label
  className="test-label c0"
  data="label"
>
  This is a label
</label>
`;

exports[`LabelLarge component should render LabelLarge snapshot 1`] = `
.c0 {
  margin-bottom: .25rem;
  font-size: 3rem;
  font-weight: 400;
  line-height: 1;
}

<label
  className="test-labelLarge c0"
  data="label-large"
>
  This is a label large
</label>
`;

exports[`Legales component should render Legales snapshot 1`] = `
.c0 {
  border-bottom: 1px solid #d5d8df;
  margin-bottom: 1rem;
  margin-top: .25rem;
  font-size: .875rem;
  font-weight: 400;
  line-height: 1.25;
}

<p
  className="test-legales c0"
  data="legales"
>
  This is a legales
</p>
`;

exports[`Logo component should render Logo snapshot 1`] = `
.c0 {
  left: 30px;
  position: absolute;
  top: 30px;
}

<img
  className="test-logo c0"
  data="logo"
/>
`;

exports[`LogoMobile component should render LogoMobile snapshot 1`] = `
.c0 {
  margin-left: 2rem;
  margin-top: 1.5rem;
}

<img
  className="test-logo-mobile c0"
  data="logo-mobile"
/>
`;

exports[`Onboarding component should render Onboarding snapshot 1`] = `
.c0 {
  background-color: #ffffff;
  overflow-y: auto;
  position: relative;
  width: 768px;
}

<div
  className="test-onboarding c0"
  data="onboarding"
>
  This is an Onboarding component
</div>
`;

exports[`SecondHeadline component should render SecondHeadline snapshot 1`] = `
.c0 {
  margin-bottom: .5rem;
  margin-top: 0;
}

<h2
  className="test-secondHeadline c0"
  data="second-headline"
>
  This is a Second Headline
</h2>
`;

exports[`SectionDescription component should render SectionDescription snapshot 1`] = `
.c0 {
  color: #ffffff;
  display: inline-block;
  font-weight: normal;
  max-width: 230px;
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  position: relative;
  z-index: 2;
}

<p
  className="test-sectionDescription c0"
  data="section-description"
>
  This is a SectionDescription component
</p>
`;

exports[`Terms component should render Terms snapshot 1`] = `
.c0 {
  border-top: 1px solid;
  font-size: .75rem;
  font-weight: 400;
  line-height: 1.33;
  padding-top: 1.25rem;
}

.c0 > p:nth-child(1) {
  font-weight: bold;
}

<div
  className="test-terms c0"
  data="terms"
>
  This is a Terms component
</div>
`;

exports[`TextPromotion component should render TextPromotion snapshot 1`] = `
.c0 {
  padding-left: 3rem;
  padding-right: 3rem;
}

<div
  className="test-text-promotion c0"
  data="text-promotion"
>
  This is a TextPromotion component
</div>
`;

exports[`Timer component should render Timer snapshot 1`] = `
.c0 {
  display: inline-block;
  padding-left: 0.25rem;
}

<span
  className="test-timer c0"
  data="timer"
>
  This is a Timer component
</span>
`;

exports[`UrlLink component should render UrlLink snapshot 1`] = `
.c0 {
  cursor: pointer;
  display: block;
  font-weight: 700;
  margin-top: 1rem;
  font-size: .875rem;
  font-weight: 700;
  line-height: 1.14;
  text-align: center;
}

<div
  className="test-urlLink c0"
  data="url-link"
>
  This is an UrlLink component
</div>
`;

exports[`Wrapper component should render Wrapper snapshot 1`] = `
.c0 {
  overflow-x: hidden;
  padding: 2rem;
  text-align: left;
}

<div
  className="test-wrapper c0"
  data="wrapper"
>
  This is a wrapper
</div>
`;
